CRM/Financial/BAO/FinancialAccount.php

```diff
@@ -224,7 +226,12 @@ public static function getARAccounts($financialAccountId, $financialAccountTypeI
    * @return int
    */
   public static function getFinancialAccountForFinancialTypeByRelationship($financialTypeID, $relationshipType) {
-    $relationTypeId = key(CRM_Core_PseudoConstant::accountOptionValues('account_relationship', NULL, " AND v.name LIKE '{$relationshipType}' "));
+    // This is keyed on the `value` column from civicrm_option_value
+    $accountRelationshipsByValue = CRM_Core_PseudoConstant::accountOptionValues('account_relationship', NULL, NULL, 'name');
+    // We look up by the name a couple times below, so flip it.
```

Comment:
I just moved this up from lower down so we only do the lookup once and just get all the name/value pairs.